addons/payment_payulatam/models/payment_acquirer.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.tools.float_utils import float_repr
+
+from hashlib import md5
+
+PAYULATAM_SUPPORTED_CURRENCIES = ['ARS', 'BRL', 'CLP', 'COP', 'MXN', 'PEN', 'USD']
+
+
+class PaymentAcquirer(models.Model):
+    _inherit = 'payment.acquirer'
+
+    provider = fields.Selection(selection_add=[
+        ('payulatam', 'PayU Latam')
+    ], ondelete={'payulatam': 'set default'})
+    payulatam_merchant_id = fields.Char(string="PayU Latam Merchant ID", required_if_provider='payulatam', groups='base.group_user')
+    payulatam_account_id = fields.Char(string="PayU Latam Account ID", required_if_provider='payulatam', groups='base.group_user')
+    payulatam_api_key = fields.Char(string="PayU Latam API Key", required_if_provider='payulatam', groups='base.group_user')
+
+    def _payulatam_generate_sign(self, inout, values):
+        if inout not in ('in', 'out'):
+            raise Exception("Type must be 'in' or 'out'")
+
+        if inout == 'in':
+            data_string = '~'.join([
+                self.payulatam_api_key,
+                self.payulatam_merchant_id,
+                values['referenceCode'],
+                float_repr(float(values['amount']), 1),
+                values['currency'],
+            ])
+        else:
+            data_string = '~'.join([
+                self.payulatam_api_key,
+                self.payulatam_merchant_id,
+                values['referenceCode'],
+                # Observation from the sandbox:
+                # PayUlatam is using a rounded value to generate their signature
+                # despite saying otherwise in the doc:
+                # http://developers.payulatam.com/en/web_checkout/integration.html
+                float_repr(float(values.get('TX_VALUE')), 1),
+                values['currency'],
+                values.get('transactionState'),
+            ])
+        return md5(data_string.encode('utf-8')).hexdigest()
+
+    def _get_payulatam_redirect_url(self):
+        self.ensure_one()
+        if self.state == 'enabled':
+            return 'https://checkout.payulatam.com/ppp-web-gateway-payu/'
+        return 'https://sandbox.checkout.payulatam.com/ppp-web-gateway-payu/'
+
+    @api.model
+    def _get_compatible_acquirers(self, *args, currency_id=None, **kwargs):
+        acquirers = super()._get_compatible_acquirers(*args, currency_id=currency_id, **kwargs)
+
+        supported_currencies = self.env['res.currency'].search([('name', 'in', PAYULATAM_SUPPORTED_CURRENCIES)])
+        if currency_id not in supported_currencies.ids:
```

Comment:
```python
currency = self.env['res.currency'].browse(currency_id)
if currency.name not in PAYULATAM_SUPPORTED_CURRENCIES:
```
Let's avoid the search and manipulate one record instead of up to 7 :).